system-contracts/contracts/precompiles/EcPairing.yul

```diff
@@ -0,0 +1,99 @@
+                ret := 192
+            }
+
+            //////////////////////////////////////////////////////////////////
+            //                      HELPER FUNCTIONS
+            //////////////////////////////////////////////////////////////////
+
+            // @dev Packs precompile parameters into one word.
+            // Note: functions expect to work with 32/64 bits unsigned integers.
+            // Caller should ensure the type matching before!
+            function unsafePackPrecompileParams(
+                uint32_inputOffsetInWords,
+                uint32_inputLengthInWords,
+                uint32_outputOffsetInWords,
+                uint32_outputLengthInWords,
+                uint64_perPrecompileInterpreted
+            ) -> rawParams {
+                rawParams := uint32_inputOffsetInWords
+                rawParams := or(rawParams, shl(32, uint32_inputLengthInWords))
+                rawParams := or(rawParams, shl(64, uint32_outputOffsetInWords))
+                rawParams := or(rawParams, shl(96, uint32_outputLengthInWords))
+                rawParams := or(rawParams, shl(192, uint64_perPrecompileInterpreted))
+            }
+
+            /// @dev Executes the `precompileCall` opcode.
+            function precompileCall(precompileParams, gasToBurn) -> ret {
+                // Compiler simulation for calling `precompileCall` opcode
+                ret := verbatim_2i_1o("precompile", precompileParams, gasToBurn)
+            }
+
+            /// @notice Burns remaining gas until revert.
+            /// @dev This function is used to burn gas in the case of a failed precompile call.
+            function burnGas() {
+                // Precompiles that do not have a circuit counterpart
+                // will burn the provided gas by calling this function.
+                precompileCall(0, gas())
+            }
+
+
+            ////////////////////////////////////////////////////////////////
+            //                      FALLBACK
+            ////////////////////////////////////////////////////////////////
+
+            let bytesSize := calldatasize()
+
+            // Check that the input is the multiple of pairs of G1 and G2.
+            if mod(bytesSize, CHUNK_SIZE_BYTES()){
+                burnGas()
+            }
+
+            let pairs := div(bytesSize, CHUNK_SIZE_BYTES())
+
+            // We conduct all validations inside the precompileCall
+
+            calldatacopy(0, 0, bytesSize)
+
+            let precompileParams := unsafePackPrecompileParams(
+                0,              // input offset in words
+                mul(6, pairs),  // input length in words multiples of (p_x, p_x, q_x_a, q_x_b, q_y_a, q_y_b)
+                0,              // output offset in words
+                1,              // output length in words (pairing check boolean)
+                pairs           // number of pairs
+            )
+            let gasToPay := ECPAIRING_GAS_COST()
```

Comment:
Gas cost should depend on the number of pairs, no?

Reply:
yep, missed that, thanks for pointing out